x-pack/plugins/triggers_actions_ui/public/application/components/builtin_action_types/xmatters/xmatters_connectors.tsx

```diff
@@ -0,0 +1,203 @@
+        <EuiFlexItem>
+          <EuiFormRow
+            id="url"
+            fullWidth
+            error={errors.url}
+            isInvalid={isUrlInvalid}
+            label={i18n.translate(
+              'xpack.triggersActionsUI.components.builtinActionTypes.xmattersAction.urlTextFieldLabel',
+              {
+                defaultMessage: 'URL',
+              }
+            )}
+          >
+            <EuiFieldText
+              name="url"
+              isInvalid={isUrlInvalid}
+              fullWidth
+              readOnly={readOnly}
+              value={url || ''}
+              data-test-subj="xmattersUrlText"
+              onChange={(e) => {
+                editActionConfig('url', e.target.value);
+              }}
+              onBlur={() => {
+                if (!url) {
+                  editActionConfig('url', '');
+                }
+              }}
+            />
+          </EuiFormRow>
+        </EuiFlexItem>
+      </EuiFlexGroup>
+      <EuiFlexGroup>
+        <EuiFlexItem>
+          <EuiSpacer size="m" />
+          <EuiTitle size="xxs">
+            <h4>
+              <FormattedMessage
+                id="xpack.triggersActionsUI.components.builtinActionTypes.xmattersAction.authenticationLabel"
+                defaultMessage="Authentication"
+              />
+            </h4>
+          </EuiTitle>
+          <EuiSpacer size="s" />
+          <EuiSwitch
+            label={i18n.translate(
+              'xpack.triggersActionsUI.components.builtinActionTypes.xmattersAction.hasAuthSwitchLabel',
+              {
+                defaultMessage: 'Require Basic Authentication for xMatters.',
+              }
+            )}
+            disabled={readOnly}
+            checked={hasAuth}
+            onChange={(e) => {
+              editActionConfig('hasAuth', e.target.checked);
+              if (!e.target.checked) {
+                editActionSecrets('user', null);
+                editActionSecrets('password', null);
+              }
+            }}
+          />
+        </EuiFlexItem>
+      </EuiFlexGroup>
+      {hasAuth ? (
+        <>
+          {getEncryptedFieldNotifyLabel(
+            !action.id,
+            2,
+            action.isMissingSecrets ?? false,
+            i18n.translate(
+              'xpack.triggersActionsUI.components.builtinActionTypes.xmattersAction.reenterValuesLabel',
+              {
+                defaultMessage:
+                  'Username and password are encrypted. Please reenter values for these fields.',
+              }
+            )
+          )}
+          <EuiFlexGroup justifyContent="spaceBetween">
+            <EuiFlexItem>
+              <EuiFormRow
+                id="xmattersUser"
+                fullWidth
+                error={errors.user}
+                isInvalid={isUserInvalid}
+                label={i18n.translate(
+                  'xpack.triggersActionsUI.components.builtinActionTypes.xmattersAction.userTextFieldLabel',
```

Comment:
you should be able to use your username.. that's what I have been using in my testing

Reply:
I did try using my username but got an authentication error. No error when using my email though.

Reply:
I am not sure why that happened. Would you like to hop on a call to see if we can figure that out?

Reply:
I am thinking maybe my user doesn't have the right roles? The error message I'm getting is `[401] Unauthorized: You are not authorized to access this integration`.

It would be great to hop on a zoom about this and maybe walk through the workflow so I understand it better.